aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2133,19 +2133,22 @@ Tensor alias(const Tensor& self) {
 Tensor unfold(const Tensor& self, int64_t dimension, int64_t size, int64_t step) {
   // some special handling to deal with allow dimension == 0 when self.dim() == 0
   dimension = at::maybe_wrap_dim(dimension, self.dim(), /*wrap_scalar=*/true);
-  int64_t max_size = self.dim() == 0 ? 1 : self.size(dimension);
+
+  const auto sizes = self.sizes();
+  const auto strides = self.strides();
+  int64_t max_size = self.dim() == 0 ? 1 : sizes[0];
```

Comment:
it used to be `sizes[dimension]`, why the change?

Reply:
Also, `fold` accepts negative dimension, so you have to do `maybe_wrap_dim`

Reply:
it's wrapped on line 2134 above (the first line of the function), isn't it?

Reply:
Oh, my bad, I'm blind.